### PR TITLE
Add Staging section to Developer Docs

### PR DESCRIPTION
**Pull request by guergana**
Bug: [T295600](https://phabricator.wikimedia.org/T295600)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -254,6 +255,13 @@ To run all the browser tests:
 sail dusk
 ```
 
+## Staging <a id="staging"></a>
+
+When there are changes that need to be tested before being deployed to production, like an UX decision or testing a new feature, the [Mismatch Finder staging server](https://mismatch-finder-staging.toolforge.org/) can be used. 
+
+To deploy to this server, push your commits to a branch starting with the name `staging/`. 
+For example `staging/my_branch_name`. The changes in the branch will be deployed to the server after all the checks have passed. If it's decided that the contents of the branch will be merged into `main`, a new Pull Request needs to be created with a name that doesn't start with the `staging` prefix and follow the approval process.
```
Comment (itamargiv) [inline docs/README.md:263]: I think we can actually also merge PRs from a staging branch, so the last sentence is not needed.

**Review by itamargiv — COMMENTED** [review]
Looks great, just one small comment and we are g2g!

File: docs/README.md (hunk 1 above)
Comment (guergana) [reply docs/README.md]: Last sentence removed. :)

**Review by itamargiv — APPROVED** [review]
thank you!

**Pull request merged**